daemon/cmd/datapath.go

```diff
@@ -458,6 +465,31 @@ func setupIPSec() (int, uint8, error) {
 	return authKeySize, spi, nil
 }
 
+func setupIPCacheVTEPMapping() error {
+	var (
+		worldID    uint32 = 2 // VTEP use world id "2"
+		encryptKey uint8  = 0 // no encrypt support
+	)
+
+	for i := 0; i < len(option.Config.VtepEndpoints); i++ {
```

Comment:
For what it's worth, rather than indexing these things everywhere, you can alternatively do:

```suggestion
	for _, ep := range option.Config.VtepEndpoints {
```

This feedback applies to all of the iteration loops in the code, and will make it easier to read.

Reply:
@joestringer this one I am not sure how to do it
so I have entry like this alignment:
```
1.1.1.1          2.2.2.2
1.1.1.0/24    2.2.2.0/24
x:x:x:x:x:x   y:y:y:y:y:y
```
then `VtepEndoint[1]` must be associated with `VtepCIDRs[1]` and `VtepMACs[1]`, how do I make sure the index is not mixed up without using index?

Reply:
Sorry, for this case you're right, given the current structure of the code it does require the index. The equivalent statement for that would be `for i, ep := range option.Config.VtepEndpoints { ...`.

While reviewing I saw this pattern in `validateVTEP()` as well and thought that the index was not necessary in those cases, we could just use the range over the slice like this. I only added this note once and happened to add it here where the index _is_ necessary.

Reply:
thanks! done